omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +274,191 @@ else if(identifierSourceType.equals(Identifier_Pool)){
         }
         return ConversionUtil.convertToRepresentation(savedIdentifierSource, Representation.DEFAULT);         
     }
+    
+    @Override
+    public Object update(String uuid, SimpleObject updateBody, RequestContext context) throws ResponseException {
+        Object updatedIdentifierSource = null;
+        Object reservedIdentifiersToUpload = updateBody.get("reservedIdentifiers");
+        Object operation = updateBody.get("operation");
+        Object generateIdentifiers = updateBody.get("generateIdentifiers");
+        IdentifierSource identifierSourceToUpdate = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid);
+        SimpleObject output = new SimpleObject();
+        if (reservedIdentifiersToUpload != null) {
+            List<String> reservedIdentifiersList = new ArrayList<String>(Arrays.asList(reservedIdentifiersToUpload.toString().split(",")));
+            if(identifierSourceToUpdate != null){
+               for (int counter = 0; counter < reservedIdentifiersList.size(); counter++) {
+                    if (StringUtils.isNotBlank(reservedIdentifiersList.get(counter))) {
+                        identifierSourceToUpdate.addReservedIdentifier(reservedIdentifiersList.get(counter));
+                    }
+                }
+                Context.getService(IdentifierSourceService.class).saveIdentifierSource(identifierSourceToUpdate);
+                return output.add("reservedIdentifiers", identifierSourceToUpdate.getReservedIdentifiers());
+            }
+        }
+        if (operation != null && operation.toString().equals("uploadFromSource")) {
+            if(identifierSourceToUpdate != null){
+                IdentifierPool pool = (IdentifierPool) identifierSourceToUpdate;
+                Object batchSize = updateBody.get("batchSize");
+                Set<PooledIdentifier> previousIdentifiers = pool.getIdentifiers();
+                if(pool != null && batchSize != null && StringUtils.isNumeric(batchSize.toString())){
+                    Context.getService(IdentifierSourceService.class).addIdentifiersToPool(pool, Integer.parseInt(batchSize.toString()));
+                }
+                Set<PooledIdentifier> newIdentifiers = pool.getIdentifiers();
+                List<PooledIdentifier> generatedIdentifiers = new ArrayList();
+                for(PooledIdentifier identifier: newIdentifiers){
+                    if(!previousIdentifiers.contains(identifier)){
+                        generatedIdentifiers.add(identifier);
+                    }
+                }
+                return output.add("identifiers", generatedIdentifiers.toString());
```

Comment:
Why are you retuning these? Are they for any later use for whoever makes the REST call?

Reply:
Yes. I display the identifiers to the user.